### PR TITLE
Bug 2058683: [4.9z] Multiple ExGW cache validation/improvements

### PR DESCRIPTION
**Pull request by trozet**
Cherry pick of: https://github.com/openshift/ovn-kubernetes/pull/895

minor conflicts were in 2nd commit:
        both modified:   go-controller/pkg/ovn/namespace.go
        both modified:   go-controller/pkg/ovn/pods.go

### CODE REVIEW

**Comment by trozet** [comment]
@tssurya PTAL

**Comment by trozet** [comment]
need https://github.com/ovn-org/ovn-kubernetes/pull/2851

File: go-controller/pkg/ovn/namespace.go [hunk 1]
```diff
@@ -79,7 +96,7 @@ func (oc *Controller) getRoutingPodGWs(nsInfo *namespaceInfo) map[string]*gatewa
 
 // addPodToNamespace adds the pod's IP to the namespace's address set and returns
 // pod's routing gateway info
-func (oc *Controller) addPodToNamespace(ns string, ips []*net.IPNet) (*gatewayInfo, map[string]*gatewayInfo, net.IP,
+func (oc *Controller) addPodToNamespace(ns string, ips []*net.IPNet) (*gatewayInfo, map[string]gatewayInfo, net.IP,
```
Comment (tssurya) [inline go-controller/pkg/ovn/namespace.go:99]: this must be the minor conflict.

**Review by tssurya — APPROVED** [review]
/lgtm

**Pull request merged**